x-either/src/X/Control/Monad/Trans/Either.hs

```diff
@@ -46,3 +51,26 @@ joinErrors g1 g2 = mapEitherT (fmap (join . bimap g2 (first g1))) . runEitherT
 --   and that the "outer" error (like a user error) becomes the Right error
 joinErrorsEither :: (Functor m, Monad m) => EitherT e (EitherT f m) a -> EitherT (Either f e) m a
 joinErrorsEither = joinErrors Right Left
+
+
+-- |
+-- Great for testing. Allows you to keep all properties in EitherT all the way to the top.
+-- Great for testIO:
+-- prop_... = testIO . runOrFail $ do <all EitherT here>
+--
+runOrFail :: (Monad m) => EitherT Text m a -> m a
+runOrFail = (=<<) (either (fail . unpack) return) . runEitherT
```

Comment:
then when testing you can compactly get your fail prop out with `unifyLeft failWith`.  my main concern with this function is ppl misusing it to drop errors silently in proper prod apps with `unifyLeft (const ())`